source/MaterialXGenOsl/OslShaderGenerator.cpp

```diff
@@ -265,16 +265,24 @@ ShaderPtr OslShaderGenerator::generate(const string& name, ElementPtr element, G
     emitFunctionCalls(graph, context, stage);
 
     // Emit final output
-    if (outputSocket->getConnection())
+    const ShaderOutput* outConnection = outputSocket->getConnection();
+    if (outConnection)
     {
-        string finalResult = outputSocket->getConnection()->getVariable();
+        string finalResult = outConnection->getVariable();
+        const string& channels = outputSocket->getChannels();
+        if (!channels.empty())
+        {
+            string preResult = finalResult;
```

Comment:
The `preResult` string should not be needed.

Reply:
I was trying to debug so needed two variables to look at temporarily. Will clean up.